src/hammer-vlsi/defaults_types.yml

```diff
@@ -0,0 +1,982 @@
+# Core configuration options and defaults for hammer-vlsi.
+# The values specified in this file are the defaults.
+# e.g. foo: "bar" in this file means that the default setting for foo is "bar".
+
+vlsi.core:
+  # Path(s) to out-of-tree hammer technology libraries.
+  technology_path: list[str]
+  technology_path_meta: str
```

Comment:
Hm, since `meta` applies to every key, is there a way to not have to say that they are always `str`s?

Reply:
I was thinking we could actually use a subset of meta-actions for the types, since a lot of keys are reused, such as `vlsi.submit`. We could extend this to the meta-actions as well.

Reply:
I'm not sure I understand... do you mean that you can use what the meta actions support as input types to determine if a key's meta action is valid? Then you wouldn't need to say that the meta key is a `str`?

Reply:
Yes, that was what I was thinking. For example, this is what is currently in the [`synthesis.submit`](https://github.com/ucb-bar/hammer/blob/master/src/hammer-vlsi/defaults.yml#L538-L542) key in `defaults.yml`:

```yaml
synthesis.submit:
  command: "${vlsi.submit.command}"
  command_meta: lazysubst
  settings: "vlsi.submit.settings"
  settings_meta: lazycrossref
```

In this suggestion, that entire block would remain unchanged between `defaults.yml` and `defaults_types.yml` and we would only need to replace the relevant substituted blocks with the corresponding types. This would give the meta-actions a use within the data type configurations, so we don't have to say they are always `str`. It would also mean more flexible configurations, since a change to a root type would mean we wouldn't have to change it across all possible children.

Reply:
I see. That is indeed one way of doing it. I don't think I understand this though: 
>It would also mean more flexible configurations, since a change to a root type would mean we wouldn't have to change it across all possible children.

Regardless, I'm really wondering whether all metas can be omitted from `defaults_types.yml` entirely? In my mind, having the type annotated for the key itself already limits what meta keys are applicable, so there's no real point having the meta keys in the type config too.

Reply:
That was actually my first thought when I created `defaults_types.yml`, as to whether to annotate the meta keys or not. It is a bit redundant given information about the key itself as well as the already limited set of meta actions. Looking at the way configs are combined, meta keys are not even considered since they are not a part of the configuration when calling  `get_config` I will remove them from the types.